tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,459 +29,467 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
+					})
+				})
+
+				When("an EtcdCluster instance is created in dryRun mode", func() {
+
+					var stdOut string
+
+					JustBeforeEach(func() {
+						stdOut = helper.CmdShouldPass("odo", "service", "create", fmt.Sprintf("%s/EtcdCluster", etcdOperator), "--dry-run", "--project", commonVar.Project)
+					})
+
+					JustAfterEach(func() {
+					})
+
+					It("should only output the definition of the CR that will be used to start service", func() {
+						helper.MatchAllInOutput(stdOut, []string{"apiVersion", "kind"})
+					})
+
+					When("the output of the command is stored in a file", func() {
+
+						var fileName string
+
+						JustBeforeEach(func() {
+							randomFileName := helper.RandString(6) + ".yaml"
+							fileName = filepath.Join(os.TempDir(), randomFileName)
+							if err := ioutil.WriteFile(fileName, []byte(stdOut), 0644); err != nil {
+								fmt.Printf("Could not write yaml spec to file %s because of the error %v", fileName, err.Error())
+							}
+						})
+
+						JustAfterEach(func() {
+							// TODO delete file
+						})
+
+						When("a service is created from the output of the dryRun command with no name", func() {
+							JustBeforeEach(func() {
+								helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName, "--project", commonVar.Project)
+							})
+
+							JustAfterEach(func() {
+								helper.CmdShouldPass("odo", "service", "delete", "EtcdCluster/example", "-f")
+							})
+
+							When("odo push is executed", func() {
+								JustBeforeEach(func() {
+									helper.CmdShouldPass("odo", "push")
+								})
+
+								JustAfterEach(func() {
+								})
+
+								It("should create pods in running state", func() {
+									// now verify if the pods for the operator have started
+									pods := oc.GetAllPodsInNs(commonVar.Project)
+									// Look for pod with example name because that's the name etcd will give to the pods.
+									etcdPod := regexp.MustCompile(`example-.[a-z0-9]*`).FindString(pods)
+
+									ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", commonVar.Project}
+									helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+										return strings.Contains(output, "Running")
+									})
+
+								})
+							})
+						})
+
+						When("a service is created from the output of the dryRun command with a specific name", func() {
+
+							var name string
+							var svcFullName string
+							JustBeforeEach(func() {
+								name = helper.RandString(6)
+								svcFullName = strings.Join([]string{"EtcdCluster", name}, "/")
+								helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName, name, "--project", commonVar.Project)
+							})
+
+							JustAfterEach(func() {
+								helper.CmdShouldPass("odo", "service", "delete", svcFullName, "-f")
+							})
+
+							When("odo push is executed", func() {
```

Comment:
Should there also be a test to list the service, list the service in json format when `odo push` is executed similar to 
```
When("an EtcdCluster instance is created with no name", func() {
    ...
    When("odo push is executed", func() {
        ...
```
?

Reply:
Yes, this structure reveals missing tests. I prefer to only refactor the tests in this PR, without adding any